main.tf

```
@@ -105,7 +105,7 @@ resource "aws_launch_template" "default" {
     http_endpoint               = (var.metadata_http_endpoint_enabled) ? "enabled" : "disabled"
     http_put_response_hop_limit = var.metadata_http_put_response_hop_limit
     http_tokens                 = (var.metadata_http_tokens_required) ? "required" : "optional"
-    http_protocol_ipv6          = (var.metadata_http_protocol_ipv6_enabled) ? "enabled" : "disabled"
+    instance_metadata_tags      = (var.metadata_instance_metadata_tags_enabled) ? "enabled" : "disabled"
```

Why remove this line?

```suggestion
    http_protocol_ipv6          = (var.metadata_http_protocol_ipv6_enabled) ? "enabled" : "disabled"
    instance_metadata_tags      = (var.metadata_instance_metadata_tags_enabled) ? "enabled" : "disabled"
```

---

This seems wrong. You're removing http_protocol_ipv6 from the aws_launch_template resource and instance_metadata_tags is also an option here

---

is not a valid metadata option, I checked the provider back to version 3, it never existed